### PR TITLE
Use latest versions of dependencies

### PR DESCRIPTION
**Pull request by meltsufin**
This upgrades the project to Hibernate Core `5.4.6.Final` and Spanner JDBC Driver `1.7.0`.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -16,7 +14,7 @@ Maven coordinates for the dialect:
 <dependency>
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-spanner-hibernate-dialect</artifactId>
-  <version> TBD </version>
+  <version>0.1.0.BUILD-SNAPSHOT</version>
```
Comment (dzou) [inline README.adoc]: To use the snapshot version, does the user need to configure a special repository to achieve this?

Comment (meltsufin) [reply README.adoc]: Not really, just Maven Central Snapshots.

Comment (meltsufin) [reply README.adoc]: This is very temporary anyway.

Comment (dzou) [reply README.adoc]: Can you mention this in the docs and then remove it once the official version is published?

**Pull request merged**